holidays/countries/south_korea.py

```diff
@@ -87,66 +77,46 @@ def __init__(self, *args, **kwargs):
         StaticHolidays.__init__(self, cls=SouthKoreaStaticHolidays)
-        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_MON)
+        kwargs.setdefault("observed_rule", SAT_SUN_TO_NEXT_WORKDAY)
+        kwargs.setdefault("observed_since", 2014)
         super().__init__(*args, **kwargs)
 
-    def _add_alt_holiday(
-        self, dt: date, name: str = "", since: int = 2014, include_sat: bool = True
-    ) -> None:
-        """
-        Add alternative holiday on first day from the date provided
-        that's not already a another holiday nor a weekend.
-
-        :param dt:
-           The date of the holiday.
-
-        :param name:
-           The name of the holiday.
-
-        :param since:
-           Year starting from which alt holiday should be added
-
-        :param include_sat:
-           Whether Saturday is to be considered a weekend in addition to
-           Sunday.
-        """
-        if not self.observed:
-            return None
-
-        target_weekday = {SUN}
-        if include_sat:
-            target_weekday.add(SAT)
-        if (dt.weekday() in target_weekday or len(self.get_list(dt)) > 1) and dt.year >= since:
-            obs_date = dt + td(days=+1)
-            while obs_date.weekday() in target_weekday or obs_date in self:
-                obs_date += td(days=+1)
-            for name in (name,) if name else self.get_list(dt):
-                if "Alternative holiday" not in name:
+    def _populate_observed(self, dts: Set[date], spec_holidays: Dict[date, str]) -> None:
+        for dt in sorted(dts.union(spec_holidays.keys())):
+            if not self._is_observed(dt):
+                continue
+            obs_date = self._get_observed_date(
+                dt, SUN_TO_NEXT_WORKDAY if dt in spec_holidays else SAT_SUN_TO_NEXT_WORKDAY
+            )
+            if obs_date != dt or len(self.get_list(dt)) > 1:
+                if obs_date == dt:
+                    obs_date = self._get_next_workday(dt)
+                names = (spec_holidays[dt],) if dt in spec_holidays else self.get_list(dt)
+                for name in names:
                     self._add_holiday(self.tr(self.observed_label) % self.tr(name), obs_date)
 
-    def _add_three_day_holiday(self, dt: date, name: str) -> None:
-        """
-        Add holiday for the date before and after the given holiday date.
-
-        :param dt:
-           The date of the holiday.
-
-        :param name:
-           The name of the holiday.
-        """
-        for dt_alt in (
-            # The day preceding %s.
-            self._add_holiday(self.tr("%s 전날") % self.tr(name), dt + td(days=-1)),
-            dt,
-            # The second day of %s.
-            self._add_holiday(self.tr("%s 다음날") % self.tr(name), dt + td(days=+1)),
-        ):
-            self._add_alt_holiday(dt_alt, name=name, include_sat=False)  # type: ignore[arg-type]
-
     def _populate_public_holidays(self):
+        def _append_observed(dt: date, since: int):
```

Comment:
As this is a nested function:
```suggestion
        def append_observed(dt: date, since: int):
```